blas-tests/tests/oper.rs

```diff
@@ -433,19 +433,19 @@ fn scaled_add_3() {
                     vec![n, q]
                 };
                 let cslice = if n == 1 {
-                    vec![SliceOrIndex::from(..).step_by(s2)]
+                    vec![AxisSliceInfo::from(..).step_by(s2)]
                 } else {
                     vec![
-                        SliceOrIndex::from(..).step_by(s1),
-                        SliceOrIndex::from(..).step_by(s2),
+                        AxisSliceInfo::from(..).step_by(s1),
+                        AxisSliceInfo::from(..).step_by(s2),
                     ]
                 };
 
                 let c = range_mat64(n, q).into_shape(cdim).unwrap();
 
                 {
                     let mut av = a.slice_mut(s![..;s1, ..;s2]);
-                    let c = c.slice(SliceInfo::<_, IxDyn>::new(cslice).unwrap().as_ref());
+                    let c = c.slice(&SliceInfo::<_, IxDyn, IxDyn>::new(cslice).unwrap());
```

Comment:
This needs to be fixed (in this PR or another), we can't design a library where `SliceInfo::<_, IxDyn, IxDyn>::new(cslice).unwrap()` is a normal thing to say.

I don't keep up with `IxDyn`.

Reply:
I just pushed a better solution. If we implement `CanSlice<IxDyn>` for `[AxisSliceInfo]`, then there's no need for the user to create a `SliceInfo` instance here; they can just write `&*cslice`. (Optionally, we could also implement `CanSlice<IxDyn>` for `Vec<AxisSliceInfo>` to avoid the need for the `*`.)

We could also consider implementing `CanSlice<IxDyn>` for `[Slice]` and `CanSlice<Dim<[usize; N]>>` for `[Slice; N]`.

I do think this test is better written by slicing individual axes with `.slice_axis*()` rather than combining the slicing information and calling `.slice()`.